mat/dense_example_test.go

```diff
@@ -128,3 +128,89 @@ func ExampleDense_Inverse() {
 	//     ⎣  0  0.5⎦
 	//
 }
+
+func ExampleDense_Mul() {
+	// Initialize two matrices, a and b.
+	a := mat.NewDense(2, 2, []float64{4, 0, 0, 4})
+	b := mat.NewDense(2, 3, []float64{4, 0, 0, 0, 0, 4})
+
+	// Take the matrix product of a and b and place the result in c.
+	var c mat.Dense
+	c.Mul(a, b)
+
+	// Print the result using the formatter.
+	fc := mat.Formatted(&c, mat.Prefix("    "), mat.Squeeze())
+	fmt.Printf("Result:\nc = %v\n\n", fc)
+	// Output:
+	// Result:
```

Comment:
I don't think we need the `// Result:` line (see that godoc already gives us something equivalent https://godoc.org/gonum.org/v1/gonum/mat#example-Dense-Add), but again this can happen in another PR.